sql-parser/src/test/java/io/crate/sql/LiteralsTest.java

```diff
@@ -47,4 +47,87 @@ public void testQuote() throws Exception {
         assertThat(Literals.quoteStringLiteral("''"), is("''''''"));
         assertThat(Literals.quoteStringLiteral("'fooBar'"), is("'''fooBar'''"));
     }
+
+    @Test
+    public void testReplaceEscapedCharsNoEscapedChars() throws Exception {
+        assertThat(Literals.replaceEscapedChars(""), is(""));
+        assertThat(Literals.replaceEscapedChars("Hello World"), is("Hello World"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsSingleCharEscapes() throws Exception {
+        assertThat(Literals.replaceEscapedChars("Hello\\bWorld"), is("Hello\bWorld"));
+        assertThat(Literals.replaceEscapedChars("Hello\\fWorld"), is("Hello\fWorld"));
+        assertThat(Literals.replaceEscapedChars("Hello\\nWorld"), is("Hello\nWorld"));
+        assertThat(Literals.replaceEscapedChars("Hello\\rWorld"), is("Hello\rWorld"));
+        assertThat(Literals.replaceEscapedChars("Hello\\tWorld"), is("Hello\tWorld"));
+        assertThat(Literals.replaceEscapedChars("\\tHelloWorld"), is("\tHelloWorld"));
+        assertThat(Literals.replaceEscapedChars("\\t\\tHelloWorld\\t"), is("\t\tHelloWorld\t"));
+        assertThat(Literals.replaceEscapedChars("\\t"), is("\t"));
+        assertThat(Literals.replaceEscapedChars("\\t\\n\\f"), is("\t\n\f"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsInvalidCharEscapes() throws Exception {
+        assertThat(Literals.replaceEscapedChars("\\"), is("\\"));
+        assertThat(Literals.replaceEscapedChars("\\s"), is("\\s"));
+        assertThat(Literals.replaceEscapedChars("\\shello"), is("\\shello"));
+        assertThat(Literals.replaceEscapedChars("hello\\s"), is("hello\\s"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsOctalByteValues() throws Exception {
+        assertThat(Literals.replaceEscapedChars("\\141"), is("a"));
+        assertThat(Literals.replaceEscapedChars("\\141\\141\\141"), is("aaa"));
+        assertThat(Literals.replaceEscapedChars("\\1411"), is("a1"));
+        assertThat(Literals.replaceEscapedChars("\\0"), is("\u0000"));
+        assertThat(Literals.replaceEscapedChars("\\07"), is("\u0007"));
+        assertThat(Literals.replaceEscapedChars("\\007"), is("\u0007"));
+    }
+
+    @Test
+    public void testReplaceEscapedCharsHexByteValues() throws Exception {
```

Comment:
Personally I'd prefer it if we could split this up into multiple tests with method names that are more specific about what aspect is being tested. A good question to ask yourself is if based on the test method name you'd be able to re-create the test method body and implement the logic that is being tested.